### PR TITLE
Part4

### PR DESCRIPTION
**Pull request by RobertShock**
I cannot figure out how to add the favorites part.
@zoeames 
@morecallan 
@rousell

### CODE REVIEW

File: javascripts/controllers/FavoritesCtrl.js [hunk 1]
```diff
@@ -3,21 +3,41 @@
...
 
-	// $scope.deleteContact = (contactId) => {
-	// 	ContactService.deletecontact(contactId).then((result) => {
-	// 		getContacts();
-	// 	}).catch((err) =>{
-	// 		console.log("error in deleteContact", err);
-	// 	});
-	// };
+	$scope.changeFav = (contact, contactId) => {
+		contact.favorite = contact.favorite ? false: true;
```
Comment (rousell) [inline javascripts/controllers/FavoritesCtrl.js]: the property here that you specify is 'favorite'. this is different from the property that you use to make the uniform contact information in your createContactObject function (in there, the property name is 'isFavorite')

**Review by rousell — COMMENTED** [review]
@RobertShock I believe that I found the issue that you are having. Try making the property name for the favorite property on your contact to be the same throughout your whole application. Given that you named the property 'favorite' in your seed data, go ahead and do the same throughout your code.

File: index.html [hunk 2]
```diff
@@ -14,12 +15,14 @@
...
     
     <script src="https://www.gstatic.com/firebasejs/4.6.0/firebase.js"></script>
+    
     <script type="text/javascript" src="./lib/node_modules/jquery/dist/jquery.min.js"></script>
     <script type="text/javascript" src="./lib/node_modules/bootstrap/dist/js/bootstrap.min.js"></script>
     <script type="text/javascript" src="./lib/node_modules/angular/angular.min.js"></script>
     <script type="text/javascript" src="./lib/node_modules/angular-route/angular-route.min.js"></script>
     <script type="text/javascript" src="./lib/node_modules/angular-animate/angular-animate.min.js"></script>
     <script type="text/javascript" src="./lib/node_modules/angular-sanitize/angular-sanitize.min.js"></script>
+
```
Comment (zoeames) [inline index.html]: delete extra blank line

File: javascripts/controllers/ContactDetailCtrl.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+app.controller('ContactDetailCtrl', function($location, $routeParams, $scope, ContactService){
+
+	$scope.contact = {};
+
+	const getContacts = () => {
+		ContactService.getSingleContact($routeParams.id).then((results) => {
+			console.log("results", results);
```
Comment (zoeames) [inline javascripts/controllers/ContactDetailCtrl.js]: remove console.log

File: javascripts/controllers/ContactDetailCtrl.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+			$scope.contact = results.data;
+		}).catch((err) => {
+			console.log('error in getSingleContact', err);
+   		});	
+	};
+
+	$scope.switchFavorite = (contact) => {
+		console.log("contact.id", contact.id);
```
Comment (zoeames) [inline javascripts/controllers/ContactDetailCtrl.js]: remove console.log

File: javascripts/controllers/EditCtrl.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+app.controller("EditCtrl", function($location, $rootScope, $routeParams, $scope, ContactService) {
+console.log("hello");
+    $scope.contact = {};
+    
+    const getContact = () => {
+        ContactService.getSingleContact($routeParams.id).then((results) => {
+            $scope.contact = results.data;
+            console.log("$scope.contact", $scope.contact);
```
Comment (zoeames) [inline javascripts/controllers/EditCtrl.js]: remove console.log

File: javascripts/controllers/EditCtrl.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+        }).catch((err) => {
+            console.log('error in getContact', err);
+        });
+    };
+
+    $scope.editSubmit = (contact) => {
+        contact.uid = $rootScope.uid;
+        console.log("contact", contact);
```
Comment (zoeames) [inline javascripts/controllers/EditCtrl.js]: remove console.log

File: javascripts/controllers/FavoritesCtrl.js [hunk 7]
```diff
@@ -1,23 +1,48 @@
...
+	$scope.deleteContact = ( contactId ) => {
+		ContactService.deleteContact(contactId).then((results) => {
+			getContacts();
+		}).catch((err) => {
+			console.log('error in deleteContact:', err);
+		});  
+	};
+
+	$scope.switchFavorite = (contact) => {
+		console.log("contact.id", contact.id);
```
Comment (zoeames) [inline javascripts/controllers/FavoritesCtrl.js]: remove console.log

File: javascripts/services/ContactService.js [hunk 8]
```diff
@@ -1,26 +1,47 @@
...
+        });
+    };
+
+    const getSingleContact = (contactId) => {
+        console.log("contactId", contactId);
+        return $http.get(`${FIREBASE_CONFIG.databaseURL}/contacts/${contactId}.json`);
+    };
+
+    const postNewContact = (newContact) => {
         console.log("newContact", newContact);
```
Comment (zoeames) [inline javascripts/services/ContactService.js]: remove console.log

File: javascripts/services/ContactService.js [hunk 9]
```diff
@@ -29,5 +50,23 @@ app.service("ContactService", function($http, $q, FIREBASE_CONFIG) {
         return $http.delete(`${FIREBASE_CONFIG.databaseURL}/contacts/${contactId}.json`);
     };
 
-    return {addNewContact, getContacts, deleteContact};
+    const updateContact = (contact, contactId) => {
+        console.log(contact, contactId);
```
Comment (zoeames) [inline javascripts/services/ContactService.js]: remove console.log

File: partials/favorites.html [hunk 10]
```diff
@@ -1,4 +1,46 @@
...
+        </div>
+    </div>
+</div>
+
+
+
+
+
+
+    <!-- <h2 class="text-center">My Favorite People</h2>
```
Comment (zoeames) [inline partials/favorites.html]: Delete any commented code - if you don't need it, it shouldn't be there

**Comment by RobertShock** [comment]
console logs removed and landing page changed to view contacts after auth

**Comment by zoeames** [comment]
👍

**Pull request merged**